File: catorce ejercicios/PHP2/Php/altura.php
<form>
    <label for="iniciales">Iniciales:</label>
    <input type="text" id="iniciales" name="iniciales"><br><br>
    <label for="altura">Altura:</label>
    <input type="text" id="altura" name="altura"><br><br>
    <button type="button" onclick="calcularAltura()">Calcular</button>
  </form>
  <div id="resultado"></div>
  
<?php
function calcularAltura() {
  $altura = $_POST['altura'];
  $iniciales = $_POST['iniciales'];

  if (!is_numeric($iniciales) || strpos($altura, "-") !== false) {
    echo "Error, ingresa datos válidos";
    return;
  }

  if ($iniciales && $altura) {
    $porciones = explode(' ', $altura);

    if (is_numeric($altura)) {
      $alturaNumero = floatval($altura);
      $metros = floor($alturaNumero);
      $centimetros = round(($alturaNumero - $metros) * 100);
      $resultado = "Tus iniciales son: " . $iniciales . " y tu altura es de " . ($metros > 0 ? $metros . " centímetro" . ($metros !== 1 ? "s" : "") : "") . ($centimetros > 0 ? ($metros > 0 ? " con " : "") . $centimetros . " centímetro" . ($centimetros !== 1 ? "s" : "") : "");
      echo $resultado;
    } else if (count($porciones) === 2 && $porciones[1] === "cm" && is_numeric($porciones[0])) {
      $centimetros = intval($porciones[0]);
      $resultado = "Tus iniciales son: " . $iniciales . " y tu altura es de " . $centimetros . " cm";
      echo $resultado;
    } else if (count($porciones) === 2 && is_numeric($porciones[0]) && is_numeric($porciones[1])) {
      $metros = intval($porciones[0]);
      $centimetros = floatval($porciones[1]);
      $resultado = "Tus iniciales son: " . $iniciales . " y tu altura es de " . $metros . " metro" . ($metros !== 1 ? "s" : "") . ($centimetros > 0 ? " con " . $centimetros . " centímetro" . ($centimetros !== 1 ? "s" : "") : "");
      echo $resultado;
    } else {
      echo "Error, ingresa datos válidos";
    }
  } else {
    echo "Error, ingresa datos válidos";
  }
}
?>

File: catorce ejercicios/PYTHON2/altura.py
def calcular_altura():
    altura = input('Altura: ')
    iniciales = input('Iniciales: ')

    if iniciales.isdigit() or "-" in altura:
        print("Error, ingresa datos válidos")
        return

    if iniciales and altura:
        porciones = altura.split(' ')

        if altura.replace('.', '', 1).isdigit():
            altura_numero = float(altura)
            metros = int(altura_numero)
            centimetros = round((altura_numero - metros) * 100)
            resultado = "Tus iniciales son: " + iniciales + " y tu altura es de " + (str(metros) + " centímetro" + ("s" if metros != 1 else "")) + (" con " + str(centimetros) + " centímetro" + ("s" if centimetros != 1 else "") if centimetros > 0 else "")
            print(resultado)
        elif len(porciones) == 2 and porciones[1] == "cm" and porciones[0].isdigit():
            centimetros = int(porciones[0])
            resultado = "Tus iniciales son: " + iniciales + " y tu altura es de " + str(centimetros) + " cm"
            print(resultado)
        elif len(porciones) == 2 and porciones[0].isdigit() and porciones[1].replace('.', '', 1).isdigit():
            metros = int(porciones[0])
            centimetros = int(float(porciones[1]) * 100)
            resultado = "Tus iniciales son: " + iniciales + " y tu altura es de " + (str(metros) + " metro" + ("s" if metros != 1 else "")) + (" con " + str(centimetros) + " centímetro" + ("s" if centimetros != 1 else "") if centimetros > 0 else "")
            print(resultado)
        else:
            print("Error, ingresa datos válidos")
    else:
        print("Error, ingresa datos válidos")

calcular_altura()
